src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -1621,14 +1621,39 @@ INamedTypeSymbol INamedTypeSymbol.ConstructedFrom
             }
         }
 
-        INamedTypeSymbol INamedTypeSymbol.Construct(params ITypeSymbol[] arguments)
+        INamedTypeSymbol INamedTypeSymbol.Construct(params ITypeSymbol[] typeArguments)
         {
-            foreach (var arg in arguments)
+            var builder = ArrayBuilder<TypeWithAnnotations>.GetInstance(typeArguments.Length);
+            foreach (var typeArg in typeArguments)
             {
-                arg.EnsureCSharpSymbolOrNull<ITypeSymbol, TypeSymbol>("typeArguments");
+                var type = typeArg.EnsureCSharpSymbolOrNull<ITypeSymbol, TypeSymbol>(nameof(typeArguments));
+                builder.Add(TypeWithAnnotations.Create(type));
+            }
+            return Construct(builder.ToImmutableAndFree(), unbound: false);
+        }
+
+        INamedTypeSymbol INamedTypeSymbol.Construct(ImmutableArray<ITypeSymbol> typeArguments, ImmutableArray<CodeAnalysis.NullableAnnotation> typeArgumentNullableAnnotations)
```

Comment:
Is there any reason to allow a default `typeArgumentNullableAnnotations`? Feels like the user is calling the wrong overload if they pass default instead of calling the other version.

Reply:
Ideally callers should be able to use a single overload.

---
In reply to: [304053732](https://github.com/dotnet/roslyn/pull/37271#discussion_r304053732) [](ancestors = 304053732)